Content.Client/_ES/Wallmount/ESWallmountVisibilitySystem.cs

```diff
@@ -0,0 +1,83 @@
+using System.Numerics;
+using Content.Shared.Throwing;
+using Content.Shared.Wall;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.Player;
+using Robust.Shared.Map;
+using Robust.Shared.Timing;
+
+namespace Content.Client._ES.Wallmount;
+
+/// <summary>
+/// This handles only showing wallmounts which are on "sides of the wall" that your eye can see,
+/// by checking their direction and rotation & comparing to their position relative to your eye.
+/// </summary>
+public sealed class ESWallmountVisibilitySystem : EntitySystem
+{
+    [Dependency] private readonly IEyeManager _eye = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly IPlayerManager _player = default!;
+    [Dependency] private readonly TransformSystem _xform = default!;
+    [Dependency] private readonly SpriteSystem _sprite = default!;
+
+    /// <inheritdoc/>
+    public override void Update(float frameTime)
```

Comment:
Should be in FrameUpdate as part of normal rendering to avoid potentially popping in.

FrameUpdate is not predicted,so the prediction check can be removed.

*Alternatively*, it is entirely possible to do all this work in an overlay (maybe just call the function on the system to do the work so you can keep all the nice helpers), as rendering is single-threaded with the game itself, which would allow this to work with other viewports.

Reply:
i deliberately didnt do frameupdate just bc i felt like there wasnt much risk in weird effects if it ran per tick instead of per frame (it looks fine, popin or whatever isnt noticeable) but i guess there isnt really a strict reason -not- to just have it in frameupdate

i dont think having this in an overlay makes sense, we aren't going to be able to render it per-viewport in a way that makes sense anyway because we have no way of changing how an entity is rendered depending on which viewport it's being rendered to/eye thats seeing it

Reply:
Hm? If it's in an overlay it can just change the sprite visibility based on the eye being used for the currently drawing viewport. It'll rerun ahead of sprites being drawn in every viewport if put in ScreenSpaceBelowWorld.

So you effectively *can* change how the entities are being rendered.